src/types/resolveDescriptors.ts

```diff
@@ -1552,6 +1550,27 @@ export function resolveDescriptors(ctx: CompilerContext, Ast: FactoryAst) {
                         `Trait "${tr.name}" requires field "${f.name}" of type "${printTypeRef(f.type)}"`,
                         t.ast.loc,
                     );
+                } else if (
+                    (f.as !== null || ex.as !== null) &&
+                    f.as !== ex.as
+                ) {
+                    if (
+                        (ex.as === "int257" && f.as === null) ||
+                        (ex.as === null && f.as === "int257")
+                    ) {
```

Comment:
let's merge all these conditions: this part is too complex and having `f.as !== null` and `f.as === null` makes me think it can be simplified